### PR TITLE
feat(docs): add documentation for PaymentCard Authorization use-case

### PR DESCRIPTION
**Pull request by koprivajakub**
#### Changelog notes 

```
* Added use-case for PaymentCard Authorization
```

#### Check during review

- [ ] JSON example extended.
  - [ ] New properties are added to the correct place in the JSON.
- [ ] New properties in the table are added to the correct place.
- [ ] Correct formatting:
  - [ ] Spacing is consistent between titles, sections, tables, ...
  - [ ] Correct JSON format - indentation.
- [ ] DateTime properties should always be defined in ISO format.

### CODE REVIEW

**Comment by koprivajakub** [comment]
closing for missing links 🤦‍♂️

File: distributor-api-v1/operations.md [hunk 1]
```diff
@@ -1263,12 +1265,12 @@ In case of an error caused by insufficient availability \(which might have decre
 }
 ```
 
-| Property | Type | Description |
-| :--- | :--- | :--- |
-| `Currency` | string | ISO 4217 code of the currency. |
-| `GrossValue` | number | Gross value of the amount. (Net + sum of TaxValues) |
-| `NetValue` | number | Net value of the amount. |
-| `TaxValues` | array of [TaxValue](operations.md#taxvalue)s | Tax values of the amount. |
+|  | Property | Type | Description |
```
Comment (vitasamek) [inline distributor-api-v1/operations.md:1268]: why is there double pipe? '|'

File: distributor-api-v1/operations.md [hunk 2]
```diff
@@ -1277,3 +1279,131 @@ In case of an error caused by insufficient availability \(which might have decre
 - `Charged` - Finite state. Payment has been successfully charged.
 - `Canceled` - Finite state. Payment has been canceled, and it has not been charged.
 - `Failed` - Finite state. Payment has not been charged.
+
+## Get Payment Cards <a id="get-payment-cards"></a>
```
Comment (vitasamek) [inline distributor-api-v1/operations.md:1283]: Why is each word in first capital format?

File: distributor-api-v1/operations.md [hunk 3]
```diff
@@ -1277,3 +1279,131 @@ In case of an error caused by insufficient availability \(which might have decre
...
+            "Id": "dc2f8608-9d71-47fd-9d41-ad1a009913c6",
+            "AuthorizationState": "Authorizable"
+        }
+    ]
+}
+```
+
+|  | Property | Type | Description |
+| :--- | :--- | :--- | :--- |
+| `PaymentCards` | array of [PaymentCard](operations.md#payment-card)s | required | Payment cards that has been requested. |
```
Comment (vitasamek) [inline distributor-api-v1/operations.md:1314]: Payment cards that has been requested. -> `The payment cards.`? It's obvious that the developer requested them :)

File: distributor-api-v1/operations.md [hunk 4]
```diff
@@ -1277,3 +1279,131 @@ In case of an error caused by insufficient availability \(which might have decre
...
+
+#### PaymentState <a id="payment-card"></a>
```
Comment (vitasamek) [inline distributor-api-v1/operations.md:1316]: incorrect title and the link. PaymentState has link to payment-card?

File: distributor-api-v1/operations.md [hunk 5]
```diff
@@ -1277,3 +1279,131 @@ In case of an error caused by insufficient availability \(which might have decre
...
+
+```json
+{
+    "Id": "dc2f8608-9d71-47fd-9d41-ad1a009913c6",
+    "AuthorizationState": "Authorizable"
+}
+```
```
Comment (vitasamek) [inline distributor-api-v1/operations.md:1323]: How is this json related to the payment State? What does it mean in this case?

File: distributor-api-v1/operations.md [hunk 6]
```diff
@@ -1277,3 +1279,131 @@ In case of an error caused by insufficient availability \(which might have decre
...
+        "ColorDepth": 24,
+        "UserAgent": "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/90.0.4430.85 Safari/537.36",
+        "Language": "cs",
+        "JavaEnabled": false,
+        "TimeZoneOffset": -120
+    }
+}
+```
+
+|  | Property | Type | Description |
```
Comment (vitasamek) [inline distributor-api-v1/operations.md:1358]: Why is there double '|' ?

File: distributor-api-v1/operations.md [hunk 7]
```diff
@@ -1277,3 +1279,131 @@ In case of an error caused by insufficient availability \(which might have decre
...
+| :--- | :--- | :--- | :--- |
+| `EnterpriseId` | string | required | Unique identifier of the enterprise in which the card is stored. |
```
Comment (vitasamek) [inline distributor-api-v1/operations.md:1360]: Enterprise is not stored in the enterprise. CC detail is stored in PCI proxy.

File: distributor-api-v1/operations.md [hunk 8]
```diff
@@ -1277,3 +1279,131 @@ In case of an error caused by insufficient availability \(which might have decre
...
+| `PaymentCardId` | string | required | Unique identifier of the payment card. |
+| `BrowserInfo` | [BrowserInfo](operations.md#browser-info) | required | Information about the users browser. |
+
+#### BrowserInfo <a id="browser-info"></a>
```
Comment (vitasamek) [inline distributor-api-v1/operations.md:1364]: Browser info

File: distributor-api-v1/operations.md (hunk 4 above)
Comment (vitasamek) [inline distributor-api-v1/operations.md:1316]: Payment state

File: distributor-api-v1/use-cases/on-session-payment-card-authorization.md [hunk 9]
```diff
@@ -0,0 +1,28 @@
...
+
+Whenever a user create reservation there is an option to let the guest insert a payment card which will be stored in Mews.
+You can read more about how to support payment cards in booking engine application in our [use-case](./how-to-support-payment-cards-in-booking-engine-application.md).
+With PSD2 directive it might be handy for the enterprise to authorize the card for smooth payments in the future. This can minimize
+the risk of a future payment being rejected.
+
+## Creating a reservation group with verified card
+
+First step of the entire workflow is to [create a reservation group via our API](../operations.md#create-reservation-group) with [payment card data](./how-to-support-payment-cards-in-booking-engine-application.md).
+The response contains a field `PaymentCardId` which we will use to check whether the card needs authorization via [requesting details of the card](../operations.md#get-payment-cards) and validating the [AuthorizationState](../operations.md#authorization-state) of the card is `Authorizable`.
```
Comment (vitasamek) [inline distributor-api-v1/use-cases/on-session-payment-card-authorization.md:11]: PaymentCardId is not saying anything about authorization for the card. AuthorizationState is saying if the CC has to be authorized.

File: distributor-api-v1/operations.md (hunk 1 above)
Comment (koprivajakub) [reply distributor-api-v1/operations.md:1268]: Just to make the file consistent
image

But next PR will be in terms of consolidation and having it this way:
image

I just did not noticed before and I would like to not mix new docs with refactoring current state

File: distributor-api-v1/operations.md (hunk 2 above)
Comment (koprivajakub) [reply distributor-api-v1/operations.md:1283]: Consistence for now:
image

But definetelly a good point also for the next PR which will be consolidation

File: distributor-api-v1/operations.md (hunk 4 above)
Comment (koprivajakub) [reply distributor-api-v1/operations.md:1316]: title should be Payment card. Fixed

File: distributor-api-v1/operations.md (hunk 5 above)
Comment (koprivajakub) [reply distributor-api-v1/operations.md:1323]: all 3 comments are on a fact where it should be `Payment card` object. Copy paste typo 🤦‍♂️

File: distributor-api-v1/operations.md (hunk 6 above)
Comment (koprivajakub) [reply distributor-api-v1/operations.md:1358]: Answered above.

File: distributor-api-v1/use-cases/on-session-payment-card-authorization.md (hunk 9 above)
Comment (koprivajakub) [reply distributor-api-v1/use-cases/on-session-payment-card-authorization.md:11]: removed the mentioned `PaymentCardId` section. I will leave it only in the step-by-step workflow section

File: distributor-api-v1/use-cases/on-session-payment-card-authorization.md [hunk 10]
```diff
@@ -0,0 +1,28 @@
...
+A `returnUrl` query parameter is optional and will be used by the application to return the user into your booking engine after successful payment card authorization or in case the user decides to abandon the flow. `returnUrl` value should be a `Base64` encoded absolute url e.g. in javascript via `btoa` function.
+After the user returns to your booking engine you can verify the card authorization state via [requesting details of the card](../operations.md#get-payment-cards) and check the [AuthorizationState](../operations.md#authorization-state) of the card is `Authorized`.
+
+## Step by step workflow
+1. [Create new reservation group](../operations.md#request-apibaseurlapidistributorv1reservationgroupscreate) with [CreditCardData](../operations.md#creditcarddata)
+2. Get `PaymentCardId` from the response
+3. Validate the card [AuthorizationState](../operations.md#authorization-state) is `Authorizable` via [Get Payment Cards](../operations.md#get-payment-cards) endpoint. **If not the workflow ends here**.
```
Comment (zdrazil) [inline distributor-api-v1/use-cases/on-session-payment-card-authorization.md:20]:
```suggestion
3. Validate that the card [AuthorizationState](../operations.md#authorization-state) is `Authorizable` via [Get Payment Cards](../operations.md#get-payment-cards) endpoint. **If not the workflow ends here**.
```

File: distributor-api-v1/use-cases/on-session-payment-card-authorization.md [hunk 11]
```diff
@@ -0,0 +1,28 @@
...
+4. Try to authorize the card via [Authorize Payment Card](../operations.md#authorize-payment-card) endpoint and check the [State](../operations.md#payment-card-authorization-state) in the response is either `Pending` or `Requested`. **If not the workflow ends here**.
```
Comment (zdrazil) [inline distributor-api-v1/use-cases/on-session-payment-card-authorization.md:21]:
```suggestion
4. Try to authorize the card via [Authorize Payment Card](../operations.md#authorize-payment-card) endpoint and check that the [State](../operations.md#payment-card-authorization-state) in the response is either `Pending` or `Requested`. **If not the workflow ends here**.
```

File: distributor-api-v1/use-cases/on-session-payment-card-authorization.md [hunk 12]
```diff
@@ -0,0 +1,28 @@
...
+5. Create a `ReturnUrl` via encoding your url by `Base64` e.g. Javascript example would be `btoa(urlWhereUserShouldReturn)`
```
Comment (zdrazil) [inline distributor-api-v1/use-cases/on-session-payment-card-authorization.md:22]: ⛏️ 
```suggestion
5. Create a `ReturnUrl` via encoding your url by `Base64` e.g. JavaScript example would be `btoa(urlWhereUserShouldReturn)`
```

**Review by zdrazil — COMMENTED** [review]
I didn't find any factual issues, just some things which could maybe improve readability.

File: distributor-api-v1/use-cases/on-session-payment-card-authorization.md [hunk 13]
```diff
@@ -0,0 +1,28 @@
...
+Next step is to check whether the card needs authorization via [requesting details of the card](../operations.md#get-payment-cards) and validating the [AuthorizationState](../operations.md#authorization-state) of the card is `Authorizable`.
+We can try to [authorize the card](../operations.md#authorize-payment-card) immediately. The response contains a [State](../operations.md#payment-card-authorization-state) of the authorization where no further action is needed unless the [State](../operations.md#payment-card-authorization-state) is not finite (`Pending` or `Requested`). In any other cases the card is either authorized without any
+user action, or the authorization is not possible. In case the [State](../operations.md#payment-card-authorization-state) is `Pending` or `Requested` you can redirect user to Mews application which will take care of the whole authorization process. Simply navigate the user to url `https://[MewsApplicationsBaseUrl]/navigator/card-authorization/detail/[PaymentCardId]?returnUrl=[ReturnUrl]` which will take care of the authorization process.
```
[the remark this answers is not in the document]
Comment (koprivajakub) [reply distributor-api-v1/use-cases/on-session-payment-card-authorization.md:13]: This kind of leads the reader to have an explicit check to `Authorized, Declined, Canceled` but they should not care about the finite states here, but rather those that are not finite. So I would keep it as it is, just with sections (paragraphs)

**Comment by koprivajakub** [comment]
@zdrazil fixed all except one where I put a comment of reasoning, hope you will agree.

**Review by zdrazil — APPROVED** [review]

**Review by vitasamek — APPROVED** [review]
